docs/developer/architecture/sso.rst

```diff
@@ -0,0 +1,126 @@
+to fallback on.
+
+Deprecate Dex
+~~~~~~~~~~~~~
+The issue of SSO is that Dex does not keep track of the session, we may as well
+get rid of Dex and replace it with other mechanism.
+The issue with this solution is that the replacement, oauth2-proxy, does not
+provide oidc functionalities and thus cannot be used by the kube-apiserver.
+
+Chosen Approach
+***************
+
+We will use a combination of Dex and Oauth2-proxy,
+
+- Dex is used to provide token from static password authentication and to proxy
+  other configured OIDC providers
+- Oauth2-proxy is used to handle authentication and session support for the
+  different endpoints
+
+Implementations details
+-----------------------
+
+Components functionalities
+**************************
+
+Dex will:
+
+- Provides static password authentication support
+- Provides tokens backed by the static password, kube-apiserver will be
+  configured to talk to Dex
+- Proxy other OIDC providers like Keycloak
+
+Oauth2-proxy will:
+
+- Be installed as a side car authentication layer for the control plane
+  components (PlatformUI, Grafana, Prometheus, etc)
+- Provide session support, moving from one components to another while keeping
+  the same session id
+- Redirect to Dex and recover the authentication token if the session does not
+  exist or no more valid
+- Provides the token to the protected upstream via headers or in the session
+  itself (some logic needed to extract the token from the session)
+
+Authentication workflow
+***********************
+
+1. User visits endpoint-1 (exp PlatformUI)
+2. Oauth2-proxy finds no session cookies, so it redirect the User to Dex
+   in order to authenticate
+3. User authenticate (with static password or through another OIDC provider)
+4. Oauth2-proxy gets a token and creates a session
+5. Oauth2-proxy passes the token to the upstream, User is logged in
+6. User visits endpoint-2 (exp Grafana)
+7. Oauth2-proxy reads the session, and redirect the User to the endpoint-2
+   while passing the token to the upstream
+8. User is automatically logged in the endpoint-2
+9. User log out, Oauth2-proxy deletes the session
+10. GOTO (1)
+
+NB: Ajax requests will receive 401 if the request does not contain the cookie.
+
+Upstream authentication
+***********************
+
+Using Oauth2-proxy, upstreams will receive the authentication token as headers.
+In case of client-side application, it should send a request to ``/userinfo``
+using the cookie and then retreive the headers from the response.
+
+The headers contain:
+
+- The user claim (X-Forwarded-User)
+- The user groups if configured (X-Forwarded-Groups)
+- The email claim (X-Forwarded-Email)
+- The full OIDC access token (X-Forwarded-Access-Token)
+
+Operating Oauth2-proxy
+**********************
+
+Oauth2-proxy serves some built-in endpoints, from which
+
+- ``/ping`` returns 200 OK if it is working correctly which can be used in
+  container healthchecks
+- ``/metrics`` provides metrics in prometheus format.
```

Comment:
We might want to add a `Monitoring Oauth2-proxy` section to define which prometheus rules we would want to deploy to be alerted in case of unavailability of Oauth2-Proxy